src/main/java/com/osiris/autoplug/client/tasks/SSHManager.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2021-2022 Osiris-Team.
+ * All rights reserved.
+ *
+ * This software is copyrighted work, licensed under the terms
+ * of the MIT-License. Consult the "LICENSE" file for details.
+ */
+
+package com.osiris.autoplug.client.tasks;
+
+import java.io.IOException;
+
+import com.osiris.autoplug.client.configs.SSHConfig;
+import com.osiris.autoplug.client.network.online.connections.SSHServerSetup;
+import com.osiris.jlib.logger.AL;
+
+public class SSHManager {
+    private static SSHManager instance;
+    private Thread sshThread;
+    private SSHServerSetup sshServerSetup;
+    private SSHConfig sshConfig;
+
+    private SSHManager(SSHConfig config) throws IOException {
+        this.sshConfig = config;
+        this.sshServerSetup = new SSHServerSetup();
+        createThread();
+    }
+
+    public static SSHManager getInstance(SSHConfig config) throws IOException {
+        if (instance == null) {
+            synchronized (SSHManager.class) {
+                if (instance == null) {
+                    instance = new SSHManager(config);
+                }
+            }
+        }
+        return instance;
+    }
+
+    private void createThread() throws IOException {
+        sshThread = new Thread(() -> {
+            try {
+                sshServerSetup.start();
+            } catch (IOException e) {
+                AL.warn("IOException occurred while starting SSH server", e);
+            } catch (Exception e) {
+                AL.warn("Failed to start SSH server", e);
+            }
+        });
+    }
+
+    public boolean start() {
```

Comment:
Add "synchronized" to this method and to the stop method, so that both cannot be executed at the same time and cause weird stuff